### PR TITLE
SAA-1178 fix activity summary view to join on correct activity and activity schedule columns.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**

### CODE REVIEW

File: src/main/resources/migrations/common/V2023.09.08__v_activity_summary_fix_join_columns.sql [hunk 1]
```diff
@@ -0,0 +1,26 @@
+CREATE OR REPLACE VIEW v_activity_summary
+AS
+select a.activity_id                                           id,
+       a.activity_category_id,
+       a.prison_code,
+       a.description                                           activity_name,
+       s.capacity,
+       (select count(*)
+        from allocation alloc
+        join activity_schedule s on s.activity_schedule_id = alloc.activity_schedule_id
```
Comment (matt-crowson-moj) [inline src/main/resources/migrations/common/V2023.09.08__v_activity_summary_fix_join_columns.sql:10]: Fix 1 here.

File: src/main/resources/migrations/common/V2023.09.08__v_activity_summary_fix_join_columns.sql [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        where s.activity_id = a.activity_id
+          and alloc.prisoner_status != 'ENDED')                allocated,
+       (select count(*)
+        from waiting_list w
+        where w.activity_id = a.activity_id
```
Comment (matt-crowson-moj) [inline src/main/resources/migrations/common/V2023.09.08__v_activity_summary_fix_join_columns.sql:15]: Fix 2 here.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/integration/PrisonIntegrationTest.kt [hunk 3]
```diff
@@ -75,7 +75,7 @@ class PrisonIntegrationTest : IntegrationTestBase() {
         activityName = "Maths Level 1",
         category = educationCategory,
         capacity = 10,
-        allocated = 2,
+        allocated = 4,
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/integration/PrisonIntegrationTest.kt:78]: You would need to look at the associated seed file to verify this.

From what I can tell looking at the seed file the number allocated count should have been 4 and not 2.

The seed file dates back to when you could have multiple schedules for an activity which turns out to be a good candidate for verifying the query changes i.e. we have a schedule where the id does not match that of the activity.  This was previously hidden due to the join on activity id being the same as the schedule id.

**Review by iterative-it — APPROVED** [review]
LGTM

**Pull request merged**